ext/json/json_encoder.c

```diff
@@ -332,6 +332,11 @@ static void php_json_escape_string(smart_str *buf, char *s, size_t len, int opti
 				smart_str_appendl(buf, "null", 4);
 				return;
 			}
+			if (us != 0x2028 && us != 0x2029 && (options & PHP_JSON_UNESCAPED_UNICODE)) {
+				pos -= 3;
+				us = (unsigned char)s[pos];
+				goto unescaped_char;
```

Comment:
think that something like this would be better:
```c
smart_str_appendl(buf, &s[pos - 3], 3);
continue; /* alternatively add the rest to else block (instead of continue...) */
```
it's untested and it's quite late here but hope you get the idea... ;)